include/cantera/thermo/PengRobinson.h

```diff
@@ -0,0 +1,365 @@
+    virtual void setToEquilState(const double* lambda_RT);
+    virtual void initThermo();
+
+    //! Retrieve a and b coefficients by looking up tabulated critical parameters
+    /*!
+    *  If pureFluidParameters are not provided for any species in the phase,
+    *  consult the critical properties tabulated in data/inputs/critProperties.xml.
+    *  If the species is found there, calculate pure fluid parameters a_k and b_k as:
+    *  \f[ a_k = 0.4278*R**2*T_c^2/P_c \f]
+    *
+    *  and:
+    *  \f[ b_k = 0.08664*R*T_c/P_c \f]
+    *
+    *  @param iName    Name of the species
+    */
+    virtual std::vector<double> getCoeff(const std::string& iName);
+
+    //! Set the pure fluid interaction parameters for a species
+    /*!
+     *  The "a" parameter for species *i* in the Peng-Robinson model is assumed
+     *  to be a linear function of temperature:
+     *  \f[ a = a_0 + a_1 T \f]
+     *
+     *  @param species   Name of the species
+     *  @param a0        constant term in the expression for the "a" parameter
+     *      of the specified species [Pa-m^6/kmol^2]
+     *  @param a1        temperature-proportional term in the expression for the
+     *      "a" parameter of the specified species [Pa-m^6/kmol^2/K]
+     *  @param b         "b" parameter in the Peng-Robinson model [m^3/kmol]
+     *  @param alpha     dimensionless function of T_r and \omega
+     *  @param omega     acentric factor
+     */
+    void setSpeciesCoeffs(const std::string& species, double a0, double b,
+                              double w);
+
+    //! Set values for the interaction parameter between two species
+    /*!
+     *  The "a" parameter for interactions between species *i* and *j* is
+     *  assumed by default to be computed as:
+     *  \f[ a_{ij} = \sqrt(a_{i, 0} a_{j, 0}) + \sqrt(a_{i, 1} a_{j, 1}) T \f]
+     *
+     *  This function overrides the defaults with the specified parameters:
+     *  \f[ a_{ij} = a_{ij, 0} + a_{ij, 1} T \f]
+     *
+     *  @param species_i   Name of one species
+     *  @param species_j   Name of the other species
+     *  @param a0          constant term in the "a" expression [Pa-m^6/kmol^2]
+     *  @param a1          temperature-proportional term in the "a" expression
+     *      [Pa-m^6/kmol^2/K]
+     */
+    void setBinaryCoeffs(const std::string& species_i,
+                         const std::string& species_j, double a0, double a1);
+
+protected:
+    // Special functions inherited from MixtureFugacityTP
+    virtual double sresid() const;
+    virtual double hresid() const;
+
+public:
+    virtual double liquidVolEst(double TKelvin, double& pres) const;
+    virtual double densityCalc(double TKelvin, double pressure, int phase, double rhoguess);
+
+    virtual double densSpinodalLiquid() const;
+    virtual double densSpinodalGas() const;
+    virtual double pressureCalc(double TKelvin, double molarVol) const;
+    virtual double dpdVCalc(double TKelvin, double molarVol, double& presCalc) const;
```

Comment:
It would be worth thinking about what methods should actually be public, versus which are just implementation details. For example, just for this set, I would imagine that `liquidVolEst`, `densityCalc`, `pressureCalc`, and `dpdVCalc` should all be protected.

Reply:
Above functions, along with a couple of other species-specific functions are now declared as Protected instead of Public.